content/en/blog/2020/large-scale-security-policy-performance-tests/index.md

```diff
@@ -0,0 +1,148 @@
+    link="./AuthZ_var_paths.svg"
+    alt="Authorization policy with variable number of paths"
+    caption=""
+    >}}
+And finally the last graph shows the data of having a variable number of `sourceIP` rules vs path rules without the baseline.
+{{< image width="90%" ratio="45.34%"
+    link="./AuthZ_paths_vs_sourceIP.svg"
+    alt="Authorization policy with both paths and `sourceIP`"
+    caption=""
+    >}}
+Having `sourceIP` rules increases the latency of request in a minimal amount compared to that of path rules.
+{{< /tab >}}
+
+{{< tab name="JWTRule" category-value="five" >}}
+{{< image width="90%" ratio="45.34%"
+    link="./RequestAuthN_jwks.svg"
+    alt="Request Authentication with variable number of `JWTRules`"
+    caption=""
+    >}}
+Having a single `JWTRule` applied to the system is comparable to that of having no policies applied, but as the number of `JWTRules` increase the latency increases disproportionately larger.
+{{< /tab >}}
+
+{{< tab name="Variable Authorization policies" category-value="six" >}}
+To test how the number of Authorization policies effect runtime the tests can be broken into two cases.
+
+- 1: Every Authorization policy has a single `sourceIP` rule.
+
+- 2: Every Authorization policy has a single path rule.
+
+{{< image width="90%" ratio="45.34%"
+    link="./AuthZ_var_policies_sourceIP.svg"
+    alt="Authorization policy with variable number of policies, with `sourceIP` rule"
+    caption=""
+    >}}
+{{< image width="90%" ratio="45.34%"
+    link="./AuthZ_var_policies_paths.svg"
+    alt="Authorization policy with variable number of policies, with path rule"
+    caption=""
+    >}}
+The overall trend of both graphs are similar. This is consistent to the paths vs `sourceIP` data which showed that the latency is marginally higher for `sourceIP` rules compared to that of path rules.
+{{< /tab >}}
+
+{{< /tabset >}}
+
+## Conclusion
+
+- In general adding security policies does not add relatively high overhead to the system, with the heavier policies being:
+
+    1. Authorization policy with `JWTRules` rules.
+
+    1. Authorization policy with `requestPrincipal` rules.
+
+    1. Authorization policy with principals rules.
+
+- In lower loads (requests with lower qps and conn) the difference in latency for most policies is minimal.
+
+- That having sidecars creates a larger increase in latency than most policies even if those policies being applied are large.
+
+- The overhead of having these extremely large policies add is relatively similar to the increase of latency of adding the Envoy proxies compared to that of no Envoy proxies.
+
+- In a more particular note, with two different tests it was determined that the `sourceIP` rule is marginally slower than that of a path rule.
+
+If you are interested in creating your own large scale Security policies and running performance tests with them see [README](https://github.com/istio/tools/tree/master/perf/benchmark/security/generate_policies)
```

Comment:
Can we also link the design doc

Reply:
Yes, my only concern is that only people from google can view the doc

Reply:
the doc should be moved to the security WG community drive, anyone can join the community group to view it.

Reply:
Great then we are all good